### PR TITLE
cache app metrics in eventgenerator

### PR DESCRIPTION
**Pull request by boyang9527**

### CODE REVIEW

File: src/autoscaler/eventgenerator/aggregator/aggregator_test.go [hunk 1]
```diff
@@ -72,7 +77,8 @@ var _ = Describe("Aggregator", func() {
 	Context("Start", func() {
 		JustBeforeEach(func() {
 			var err error
-			aggregator, err = NewAggregator(logger, clock, testAggregatorExecuteInterval, testSaveInterval, appMonitorsChan, getPolicies, fakeStatWindowSecs, appMetricChan, appMetricDatabase)
+			aggregator, err = NewAggregator(logger, clock, testAggregatorExecuteInterval, testSaveInterval, appMonitorsChan,
+				getPolicies, saveAppMetricToCache, fakeStatWindowSecs, appMetricChan, appMetricDatabase)
 			Expect(err).NotTo(HaveOccurred())
 			aggregator.Start()
 			Expect(appMetricChan).Should(BeSent(&models.AppMetric{
```
[the remark this answers is not in the document]
Comment (boyang9527) [reply src/autoscaler/eventgenerator/aggregator/aggregator_test.go]: added

**Comment by qibobo** [comment]
@boyang9527 there are conflicts.

File: src/autoscaler/eventgenerator/generator/evaluator.go [hunk 2]
```diff
@@ -151,14 +154,22 @@ func (e *Evaluator) doEvaluate(triggerArray []*models.Trigger) {
 
 func (e *Evaluator) retrieveAppMetrics(trigger *models.Trigger) ([]*models.AppMetric, error) {
 	queryEndTime := time.Now()
-	queryStartTime := queryEndTime.Add(0 - 2*trigger.BreachDuration())
+
+  queryStartTime := queryEndTime.Add(0 - 2*trigger.BreachDuration())
```
Comment (cdlliuy) [inline src/autoscaler/eventgenerator/generator/evaluator.go]: indentation is incorrect here.

File: api/npm-shrinkwrap.json [hunk 3]
```diff
@@ -1132,7 +1132,7 @@
     "helmet": {
       "version": "3.9.0",
       "resolved": "https://registry.npmjs.org/helmet/-/helmet-3.9.0.tgz",
-      "integrity": "sha512-czCyS77TyanWlfVSoGlb9GBJV2Q2zJayKxU5uBw0N1TzDTs/qVNh1SL8Q688KU0i0Sb7lQ/oLtnaEqXzl2yWvA==",
+      "integrity": "sha1-eyzwFaLRCbyoPt55JEIHmcDmfe4=",
```
Comment (cdlliuy) [inline api/npm-shrinkwrap.json]: Why included the changes for  api and service broker component in this PR?

Comment (boyang9527) [reply api/npm-shrinkwrap.json]: they were updated when running integration test most likely I upgraded my local node and npm.

**Comment by cdlliuy** [comment]
LGTM

**Pull request merged**